### PR TITLE
Adding functionality for dumping the codex out as a servable webpage.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Changes codex font colours to be CSS classes.
- Adds a proc that will dump the entire codex out as an interlinked website.

## Why and what will this PR improve
People have been requesting an out-of-game resource and making it a dump of the codex will:
- Encourage people to improve the display and contents of the codex itself.
- Prevent things getting out of sync with the codebase a la the Bay wiki.

## Authorship
Me and @Lohikar so far.

## TODO
- [X] Get basic dump and link parsing working.
- [x] Generate a navbar and insert into pages.
- [x] Generate category footers and insert into relevant pages.
- [x] Clean up the CSS and add spans where appropriate so the output can be formatted nicely.
- [x] Make the CSS look nice :(
- [ ] Generate disambiguation pages.
- [ ] Implement a searchbar.
- [x] Add a unit test for broken links and correct any outstanding (`fool's gold` and `the codex` are both broken)
- [x] Test how the folder handles if dumped into Apache or something.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Depends on #2687

File: code/_helpers/text.dm [hunk 1]
```diff
@@ -319,6 +319,25 @@
 /proc/trim(text)
 	return trim_left(trim_right(text))
 
+// Remove starting and ending <br> and variants
+/proc/trim_trailing_linebreaks(var/text)
```
Comment (Lohikar) [inline code/_helpers/text.dm]: This should probably be replaced with the regex solution, if only to only do one pass through the string.

Comment (MistakeNot4892) [reply code/_helpers/text.dm]: #2687 adds this code, has also been replaced by a regex on that PR. Will rebase this one after it's approved/merged.

**Comment by MistakeNot4892** [comment]
CSS depends on https://github.com/NebulaSS13/Nebula/pull/2739

**Comment by MistakeNot4892** [comment]
Going to close this for now just to get it off the list.

**Pull request closed without merging**